Related to T936: update the show tech-support scripts

shafiev opened the pull request:
Added network related mode output
Added system - os related (cpu, memory, file system space, pkgs etc.) related mode output

shafiev committed:
Related to T936: update the show tech-support scripts
Added network related mode output
Added system - os related (cpu, memory, file system space, pkgs etc.) related mode output

dmbaturin commented on functions/tech-support:
```diff
@@ -68,241 +70,256 @@ fi
 header CONFIGURATION 
+  
+function show_version { 
+  if cli-shell-api validateTmplPath entitlement; then
```
This is one of the cases where cleanup was needed badly. The "entitlement" commands haven't existed in years.

shafiev replied:
I have deleted this block

dmbaturin commented on functions/tech-support:
```diff
@@ -68,241 +70,256 @@ fi
+    show entitlement
+  fi
 
-if cli-shell-api validateTmplPath entitlement; then
```
Mixing formatting changes with logic changes is a very bad style. How is the reviewer supposed to follow the logic changes when the file is littered with inconsequential formatting changes?

shafiev replied:
Delete this block

dmbaturin commented on functions/tech-support:
```diff
@@ -314,257 +331,299 @@ if cli-shell-api existsActive protocols bgp; then
-  echo "Wan Load Balance is not configured"
-fi
+  header VRRP
+  # XXX: not checking if configured, we'd have to walk all VIFs
```
A long solved problem: VRRP is in "high-availability vrrp".

shafiev replied:
Maybe replace is with show vrrp detail command?

dmbaturin replied:
Oh right, removing a comment that describes a problem solves the problem.
You can see that before gathering outputs from every component, the script tries to find out if it's configured to begin with. Like in `if cli-shell-api existsActive protocols ospf;`. 

VRRP doesn't fit that pattern. The comment vaguely explains why: it was hard to do. I clearly say why it's no longer the case: VRRP is a single top level path "high-availability vrrp".

shafiev replied:
I have added checking for VRRP  .

dmbaturin commented on functions/tech-support:
```diff
@@ -314,257 +331,299 @@ if cli-shell-api existsActive protocols bgp; then
+    header DNS Forwarding Statistics
+    show dns forwarding statistics
+    header DNS Forwarding Nameservers  
+    show dns forwarding nameservers
```
This command doesn't exist anymore.

shafiev replied:
Deleted this command

dmbaturin commented on functions/tech-support:
```diff
@@ -314,257 +331,299 @@ if cli-shell-api existsActive protocols bgp; then
+    header IPSec sa
+    show vpn ipsec sa
+    header IPSec sa Detail
+    show vpn ipsec sa detail
```
This command no longer exists.

shafiev replied:
Delete this command

dmbaturin replied:
So you've kept all "run show vpn ipsec sa *" commands that no longer exist and removed the one that does? https://github.com/vyos/vyatta-op/blob/9d6c8e3426145818802fc62e75168925bc4323cf/functions/tech-support#L416-L422

Or I'm misunderstanding your intent?

shafiev replied:
Please check .There was a just some misunderstanding, sorry

dmbaturin commented on functions/tech-support:
```diff
@@ -314,257 +331,299 @@ if cli-shell-api existsActive protocols bgp; then
+    header IPSec sa Detail
+    show vpn ipsec sa detail
+    header IPSec sa Statistics
+    show vpn ipsec sa statistics
```
This command no longer exists, which would be trivial to find out by running it.

shafiev replied:
Delete this command

shafiev committed:
Remove doesn't exists command entitlement dns
                              forwarding nameservers
			      dns forwarding nameservers
		              vpn ipsec sa detail
			      vpn ipsec sa statistics
Remove VRRP block comment because as far as i understood is it a long solved problem

shafiev committed:
Added check for vrrp related node in config tree
Last only needed output for network and system option mode

shafiev committed:
Remove outdated ipsec sa  command
